core/templates/dev/head/services/speech-synthesis-chunker.service.spec.ts

```diff
@@ -32,87 +32,212 @@ describe('Speech Synthesis Chunker Service', () => {
+      );
+
+      expect(speechSynthesisChunkerService.convertToSpeakableText(html))
+        .toBe('5 minus 1 Speech. Text. ');
+    });
+
+    it('should properly convert the text-with-value attribute to' +
+      ' speakable text', () => {
+      const html = (
+        '<oppia-noninteractive-link text-with-value="&quot;">' +
+        '</oppia-noninteractive-link>' +
+        '<li>"Speech"</li>' +
+        '<li>Text</li>'
+      );
+
+      expect(speechSynthesisChunkerService.convertToSpeakableText(html))
+        .toBe('Speech. Text. ');
+    });
+  });
+
+  describe('speak', function() {
+    const MockSpeechSynthesisUtteranceConstructor = (
+      SpeechSynthesisUtterance);
+    const mockSpeechSynthesisUtteran = {
+      speak: () => {},
+      onend: () => {}
+    };
+
+    beforeEach(() => {
+      spyOn(window, 'SpeechSynthesisUtterance').and.returnValue(
+        // @ts-ignore mock doesn't have all property and methods of a native
+        // SpeechSynthesisUtterance
+        mockSpeechSynthesisUtteran);
+    });
+
+    it('should not speak when chunk is too short', () => {
+      const speakSpy = spyOn(window.speechSynthesis, 'speak').and
+        .callFake(function() {
+          mockSpeechSynthesisUtteran.onend();
+        });
+      const speechSynthesisUtterance = (
+        new MockSpeechSynthesisUtteranceConstructor('a'));
+      const callbackSpy = jasmine.createSpy('callback');
+      speechSynthesisChunkerService.speak(
+        speechSynthesisUtterance, callbackSpy);
+
+      expect(callbackSpy).toHaveBeenCalled();
+      expect(speakSpy).not.toHaveBeenCalled();
+    });
+
+    it('should not speak when chunk is a falsy value', () => {
+      const speakSpy = spyOn(window.speechSynthesis, 'speak').and
+        .callFake(function() {
+          mockSpeechSynthesisUtteran.onend();
+        });
+      const speechSynthesisUtterance = (
+        new MockSpeechSynthesisUtteranceConstructor(''));
+      const callbackSpy = jasmine.createSpy('callback');
+      speechSynthesisChunkerService.speak(
+        speechSynthesisUtterance, callbackSpy);
+
+      expect(callbackSpy).toHaveBeenCalled();
+      expect(speakSpy).not.toHaveBeenCalled();
+    });
+
+    it('should speak speech twice', () => {
+      // For some reason, just call install doesn't work, we need to uninstall
+      // first so then actually install.
+      jasmine.clock().uninstall();
+      jasmine.clock().install();
+
+      const speakSpy = spyOn(window.speechSynthesis, 'speak').and
+        .callFake(function() {
+          mockSpeechSynthesisUtteran.onend();
+        });
+      const speechSynthesisUtterance = (
+        new MockSpeechSynthesisUtteranceConstructor(
+          'Value inside utterance for testing purposes.' +
+          ' This is the next chunk'));
+      const callbackSpy = jasmine.createSpy('callback');
+      speechSynthesisChunkerService.speak(
+        speechSynthesisUtterance, callbackSpy);
+
+      jasmine.clock().tick(5000);
```

Comment:
is this adding a delay in the test and what's 500, is it msec? (If yes then why so considering we have added `speak` method.)

/cc @kevinlee12 Just wanted to check whether it's fine to add a delay in the frontend test?

Reply:
Sorry, I don't get your point. The tick is for 5 secs.

Reply:
It's better to not delay the test. If what you are testing is  an async function, can you bind your test to a listener?

Reply:
@marianazangrossi, CAn you please reply to Kevin's comment?

Reply:
I've replied it, Thanks @DubeySandeep

Reply:
Done.